pkg/shipper/shipper.go

```diff
@@ -314,13 +329,15 @@ func (s *Shipper) Sync(ctx context.Context) (uploaded int, err error) {
 		// Skip overlap check if out of order uploads is enabled.
 		if m.Compaction.Level > 1 && !s.allowOutOfOrderUploads {
 			if err := checker.IsOverlapping(ctx, m.BlockMeta); err != nil {
-				return 0, errors.Errorf("Found overlap or error during sync, cannot upload compacted block, details: %v", err)
+				s.metrics.dirSyncFailures.Inc()
+				return uploaded, errors.Errorf("Found overlap or error during sync, cannot upload compacted block, details: %v", err)
 			}
 		}
 
 		if err := s.upload(ctx, m); err != nil {
 			if !s.allowOutOfOrderUploads {
-				return 0, errors.Wrapf(err, "upload %v", m.ULID)
+				s.metrics.dirSyncFailures.Inc()
```

Comment:
Is it easier to update this metric in a defer? We can increment the metric if there is any error

Reply:
Makes sense. Gonna do the change. Thanks